Nil pointer deref when using RowsAffected/LastInsertId with Copy in

Minimal reproducer:

```go
package main

import (
	"context"
	"database/sql"

	"github.com/lib/pq"
)

func main() {
	db, err := sql.Open("postgres", "host=localhost port=32768 user=postgres sslmode=disable")
	if err != nil {
		panic(err)
	}

	ctx := context.Background()

	txn, err := db.BeginTx(ctx, nil)
	if err != nil {
		panic(err)
	}

	copyStmt, err := txn.PrepareContext(ctx, pq.CopyIn("voter", "first_name", "last_name"))
	if err != nil {
		panic(err)
	}

	res, err := copyStmt.ExecContext(ctx)
	if err != nil {
		panic(err)
	}

	if res != nil {
		res.RowsAffected()
		res.LastInsertId()
	}
}
```

And the resulting crash:

```
alex@penguin /t/x> go run t.go 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4c81a3]

goroutine 1 [running]:
database/sql.driverResult.RowsAffected(0x6bef20, 0xc0000f6280, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/database/sql/sql.go:3085 +0x73
main.main()
	/tmp/x/t.go:34 +0x1fb
exit status 2
```

I'm happy to send a patch for this if someone can suggest what the right place to look in the codebase is.

--- a/conn_test.go
+++ b/conn_test.go
@@ -1739,3 +1739,31 @@ func TestMultipleResult(t *testing.T) {
 		t.Fatal("incorrect number of rows returned")
 	}
 }
+
+func TestCopyInStmtAffectedRows(t *testing.T) {
+	db := openTestConn(t)
+	defer db.Close()
+
+	_, err := db.Exec("CREATE TEMP TABLE temp (a int)")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	txn, err := db.BeginTx(context.TODO(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	copyStmt, err := txn.Prepare(CopyIn("temp", "a"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	res, err := copyStmt.Exec()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	res.RowsAffected()
+	res.LastInsertId()
+}

--- a/copy.go
+++ b/copy.go
@@ -229,7 +229,7 @@ func (ci *copyin) Exec(v []driver.Value) (r driver.Result, err error) {
 	}
 
 	if len(v) == 0 {
-		return nil, ci.Close()
+		return driver.RowsAffected(0), ci.Close()
 	}
 
 	numValues := len(v)

